### PR TITLE
Code Review - remove expiration logic and fix eyeball links

### PR DESCRIPTION
**Pull request by maureensturgeon**
- Removes references to expired version logic for code review, instead we're going to create an error page for project versions that could not be found
- Removes project expires at column from code review table
- Hides eyeball link to old versions on code reviews other than your own

Your own code review:
Screen Shot 2022-06-03 at 11 00 26 AM

Peers code review:
Screen Shot 2022-06-03 at 11 01 02 AM

Background on expiration changes: it's difficult to reliably tell if a version has expired, so we've decided that instead we are going to make a better error page for expired versions and leave links as available forever. The chances of a student looking back on a code review from over a year ago and clicking on an old version of the code is very unlikely anyway, and if they do, they will get an error page. 

## Links
- jira ticket: [LP-2393](https://codedotorg.atlassian.net/browse/LP-2393)

## Testing story
Tested locally, updated unit tests

## Deployment strategy
Deployed behind code_review_v2 flag

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx [hunk 1]
```diff
@@ -15,23 +16,25 @@ export const codeReviewTimelineElementType = {
...
-  isProjectVersionExpired,
+  viewAsCodeReviewer,
   children
 }) => {
-  const displayVersion = !isProjectVersionExpired && !!projectVersionId;
   const versionLink =
     location.origin + location.pathname + '?version=' + projectVersionId;
 
+  // You can only see previous versions of your own project
+  const displayEyeball = !viewAsCodeReviewer && !!projectVersionId;
```
Comment (jamescodeorg) [inline apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx:36]: This is kind of a side question.  Is the intent here that the code reviewer can only see the version that is being reviewed?  If so, are we achieving this by loading the version associated with code review?  Or by loading the latest version and enforcing that the code is read-only while the review is open?

@moneppo, if the code reviewer can't view other versions of the code, why do we show them the timeline?

File: dashboard/db/schema.rb [hunk 2]
```diff
@@ -2206,17 +2205,6 @@
...
-  add_foreign_key "census_inaccuracy_investigations", "census_overrides"
-  add_foreign_key "census_inaccuracy_investigations", "census_submissions"
-  add_foreign_key "census_inaccuracy_investigations", "users"
-  add_foreign_key "census_overrides", "schools"
-  add_foreign_key "census_submission_form_maps", "census_submissions"
-  add_foreign_key "census_summaries", "schools"
-  add_foreign_key "circuit_playground_discount_applications", "schools"
-  add_foreign_key "hint_view_requests", "users"
-  add_foreign_key "ib_school_codes", "schools"
-  add_foreign_key "level_concept_difficulties", "levels"
```
Comment (jamescodeorg) [inline dashboard/db/schema.rb]: These changes look unrelated?

File: dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class CodeReviewRemoveExpiresAt < ActiveRecord::Migration[6.0]
```
Comment (jamescodeorg) [inline dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb]: Do you want to apply this separately, in case we need to revert the rest of the change?  (Up to you.)

**Review by jamescodeorg — APPROVED** [review]

File: apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx (hunk 1 above)
Comment (maureensturgeon) [reply apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx:36]: Yeah, I can't find the thread where we talked about it, so maybe we chatted in a meeting, but it was decided that a student shouldn't be allowed to view the version history of another student. That's a good point James, that maybe the timeline UI doesn't make sense in that case. Or we allow student's to view older versions of another student's code.

File: dashboard/db/schema.rb (hunk 2 above)
Comment (maureensturgeon) [reply dashboard/db/schema.rb]: Whoops, yes, thanks

File: dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb (hunk 3 above)
Comment (maureensturgeon) [reply dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb]: Sure, I'll remove this because I need to update schema.rb anyway

**Review by molly-moen — APPROVED** [review]

File: dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb (hunk 3 above)
Comment (maureensturgeon) [reply dashboard/db/migrate/20220603175343_code_review_remove_expires_at.rb]: https://github.com/code-dot-org/code-dot-org/pull/46706

File: apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx (hunk 1 above)
Comment (jamescodeorg) [reply apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx:36]: @moneppo I'll take this opportunity to say that, in the future for a feature like this, we should have separate mocks for each "role" (in this case, code review owner, peer, and teacher).  I think it would help us discover these types of issues earlier in the process and reduce the overall development cost.
